public/app/plugins/datasource/azuremonitor/resourcePicker/resourcePickerData.ts

```diff
@@ -359,28 +357,40 @@ export default class ResourcePickerData extends DataSourceWithBackend<AzureMonit
   private async fetchAllNamespaces() {
     const subscriptions = await this.getSubscriptions();
     reportInteraction('grafana_ds_azuremonitor_subscriptions_loaded', { subscriptions: subscriptions.length });
-    let supportedMetricNamespaces: string[] = [];
-    for await (const subscription of subscriptions) {
+
+    let supportedMetricNamespaces: Set<string> = new Set();
+    // We make use of these three regions as they *should* contain every possible namespace
```

Comment:
Is there no endpoint from Azure for these regions?

Reply:
There's no endpoint provided by Azure that we can call that will provide us with a definitive list of all metric namespaces. The compromise is calling the general metric namespace endpoint against these 3 regions _should_ return the vast majority of namespaces.